source/exe/win32/platform_impl.cc

```diff
@@ -1,17 +1,60 @@
+#include <chrono>
+#include <thread>
+
+#include "common/buffer/buffer_impl.h"
 #include "common/common/assert.h"
 #include "common/common/thread_impl.h"
+#include "common/event/signal_impl.h"
 #include "common/filesystem/filesystem_impl.h"
 
 #include "exe/platform_impl.h"
 
 namespace Envoy {
 
+static volatile bool shutdown_pending = false;
```

Comment:
On posix this is supposed to be `sig_atomic_t` I think. Is that available on windows? `volatile` seems like the wrong qualifier here.

Reply:
looking into it.

Reply:
fixed by making it an atomic